packages/fxa-settings/src/components/PageAvatar/index.tsx

```diff
@@ -0,0 +1,332 @@
+import { useAlertBar, useMutation } from '../../lib/hooks';
+
+import AlertBar from '../AlertBar';
+import Avatar from '../Avatar';
+import FlowContainer from '../FlowContainer';
+import {
+  AddPhotoBtn,
+  ConfirmBtns,
+  RemovePhotoBtn,
+  RotateBtn,
+  TakePhotoBtn,
+  ZoomInBtn,
+  ZoomOutBtn,
+} from './buttons';
+
+export const UPDATE_AVATAR_MUTATION = gql`
+  mutation updateAvatar($file: Upload!) {
+    updateAvatar(input: { file: $file }) {
+      clientMutationId
+      avatarUrl
+    }
+  }
+`;
+
+const frameClass = `rounded-full m-auto w-40 object-cover`;
+
+export const PageAddAvatar = (_: RouteComponentProps) => {
+  const navigate = useNavigate();
+  const account = useAccount();
+  const { l10n } = useLocalization();
+  const { avatarUrl } = useAccount();
+  const alertBar = useAlertBar();
+  const [saveEnabled, setSaveEnabled] = useState(false);
+
+  /* Capture State */
+
+  const webcamRef = useRef<Webcam>(null);
+  const [capturedImgSrc, setCapturedImgSrc] = useState<undefined | string>(
+    undefined
+  );
+  const [camLoaded, setCamLoaded] = useState(false);
+  const [capturing, setCapturing] = useState(false);
+  const [mediaError, setMediaError] = useState(false);
+
+  /* Edit State */
+
+  const [crop, setCrop] = useState({ x: 0, y: 0 });
+  const [zoom, setZoom] = useState(1);
+  const [rotation, setRotation] = useState(0);
+  const [croppedImgSrc, setCroppedImgSrc] = useState<null | Blob>(null);
+  const [croppedAreaPixels, setCroppedAreaPixels] = useState({
+    width: 0,
+    height: 0,
+    x: 0,
+    y: 0,
+  });
+
+  const resetAllState = () => {
+    setZoom(1);
+    setRotation(0);
+    setCrop({ x: 0, y: 0 });
+    setCroppedImgSrc(null);
+    setCapturedImgSrc(undefined);
+    setSaveEnabled(false);
+  };
+
+  const [updateAvatar] = useMutation(UPDATE_AVATAR_MUTATION, {
+    onCompleted: () => {
+      firefox.profileChanged(account.uid);
+      navigate(HomePath, { replace: true });
+    },
+    onError: onFileError,
+    update: (cache, { data: { updateAvatar } }) => {
+      cache.modify({
+        id: cache.identify({ __typename: 'Account' }),
+        fields: {
+          avatarUrl() {
+            return updateAvatar.avatarUrl;
+          },
+        },
+      });
+    },
+  });
+
+  /* Edit Handlers */
+
+  const onCropComplete = useCallback(
+    (_croppedArea, croppedAreaPixels) => {
+      setCroppedAreaPixels(croppedAreaPixels);
+      setSaveEnabled(true);
+    },
+    [setCroppedAreaPixels]
+  );
+
+  const saveCroppedImage = useCallback(async () => {
+    try {
+      if (capturedImgSrc && croppedAreaPixels) {
+        const croppedImage = await getCroppedImg(
+          capturedImgSrc,
+          croppedAreaPixels,
+          rotation
+        );
+        updateAvatar({
+          variables: { file: croppedImage },
+        });
+      } else throw new Error('no image data');
+    } catch (e) {
+      console.error(e);
+    }
+  }, [croppedAreaPixels, rotation, capturedImgSrc, updateAvatar]);
```

Comment:
I think we could refactor this and `save` below to be a little simpler, but it's not a blocker.